account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -120,8 +129,36 @@ export async function createSMAV2Account(
     ),
   });
 
+  // TODO: add deferred action flag
+  const getAccountNonce = async (nonceKey?: bigint): Promise<bigint> => {
+    // uint32 entityId + (bytes1 options) makes a uint40
+    const nonceKeySuffix: bigint =
+      entityId * 256n + (isGlobalValidation ? 1n : 0n);
+
+    if (nonceKey) {
```

Comment:
💬  Please don't change anything, I just want a discussion.
 I also tend to see mutation in TS and think of a fn.
```ts
const _globalNonce = (nonceKey?: bigint) => {
const nonceKeySuffix: bigint =
      entityId * 256n + (isGlobalValidation ? 1n : 0n);
      if (nonceKey) {
         if (nonceKey % 2n ** 40n !== nonceKeySuffix) {
            throw new Error("Invalid nonceKey");
          }
      }
      return nonceKeySuffix
```

Then in the entryPointContract, which is the only place that we use the nonce
```ts
return entryPointContract.read.getNonce([
      accountAddress,
      __globalNonce(nonceKey),
    ]) as Promise<bigint>;
```
Or something like that. In TS I have an aversion to setting variables, yet in rust I do it all the time. I don't know the correct solution.

Reply:
Interesting approach. I don't think there is a correct solution in general, but I have a feeling that some of the code here would change once @moldy530 takes a look :P

Reply:
I also like the solution that @jaypaik has as well with ?? https://github.com/alchemyplatform/aa-sdk/pull/1209#discussion_r1883340174, it also works in the similar vain of avoid re-assigning a variable.